### PR TITLE
Feature5 send accrual

### PR DESCRIPTION
**Pull request by lingaoliu105**
For demonstrating feature 5: can adjust the auto sending and retrieving time sooner, demonstrate the program and show drivehq content

How to Demo?
1. Go to accrual processor. 
2. user perform_later
3. Test the upload (download does not work for now)
4. AccrualProcessor.convert_to_accrual(tn)

### CODE REVIEW

File: app/jobs/send_accrual_job.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class SendAccrualJob < ApplicationJob
+  config.active_job.queue_adapter = :sidekiq
```
Comment (ExtremelySunnyYK) [inline app/jobs/send_accrual_job.rb:2]: error here

File: app/jobs/send_accrual_job.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+  queue_as :default
+
+  def perform(*args)
+    # Do something later
+    puts "task performing"
+
+    system "cmdftp -s:drivehq_send.txt -%filepath:#{args[0]}"
```
Comment (ExtremelySunnyYK) [inline app/jobs/send_accrual_job.rb:10]: what does this line do?

Comment (ExtremelySunnyYK) [reply app/jobs/send_accrual_job.rb:10]: it to send to send to sftp, we need to do this

File: app/jobs/send_accrual_job.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class SendAccrualJob < ApplicationJob
+
+  queue_as :default
+
+  def perform(*args)
+    # Do something later
+    puts "task performing"
+
+    system "cmdftp -s:drivehq_send.txt -%filepath:#{args[0]}"
```
Comment (ExtremelySunnyYK) [inline app/jobs/send_accrual_job.rb:9]: how do we know if this work?

File: app/controllers/loyalty_program_data_controller.rb [hunk 4]
```diff
@@ -58,9 +58,10 @@ def update
 
 
  
-        Transaction.create(amount: loyalty_program_datum_params[:in_points].to_d, loyalty_program_data_id: params[:id], status: 0, account_id: current_user.id)
-        format.html { redirect_to loyalty_program_datum_url(@loyalty_program_datum), notice: "Point Transfer is now being processed." }
-        # format.json { render :show, status: :ok, location: @loyalty_program_datum }
+        @transaction = Transaction.create(amount: loyalty_program_datum_params[:points].to_d, loyalty_program_data_id: params[:id], status: 0, account_id: current_user.id)
```
Comment (ExtremelySunnyYK) [inline app/controllers/loyalty_program_data_controller.rb:61]: so is this loyalty_program_data_id? or?

Comment (ExtremelySunnyYK) [reply app/controllers/loyalty_program_data_controller.rb:61]: loyalty_program_datum_id

File: app/models/transaction.rb [hunk 5]
```diff
@@ -3,17 +3,11 @@ class Transaction < ApplicationRecord
 
   validate :amount
 
-  # belongs_to :loyalty_program_data, class_name: "loyalty_program_datum", foreign_key: "loyalty_program_datum_id"
+  belongs_to :loyalty_program_datum
```
[the remark this answers is not in the document]
Comment (mattChrisP) [reply app/models/transaction.rb:6]: I think should be ok

**Comment by ExtremelySunnyYK** [comment]
image
Error over here

**Review by ExtremelySunnyYK — CHANGES_REQUESTED** [review]
fix errors and try on your local machine. Thanks

**Pull request merged**